server/src/main/java/com/linecorp/centraldogma/server/internal/api/ProjectServiceV1.java

```diff
@@ -83,9 +87,47 @@ public CompletableFuture<List<ProjectDto>> listProjects(@Param @Nullable String
                                                                         .collect(toImmutableList()), executor);
         }
 
-        return CompletableFuture.supplyAsync(() -> projectApiManager.listProjects().values().stream()
-                                                                    .map(DtoConverter::convert)
-                                                                    .collect(toImmutableList()), executor);
+        return CompletableFuture.supplyAsync(() -> {
+            return projectApiManager.listProjects().values().stream()
+                                    .map(project -> DtoConverter.convert(project, getUserRole(project, user)))
+                                    .collect(toImmutableList());
+        }, executor);
+    }
+
+    private static ProjectRole getUserRole(Project project, User user) {
+        if (user.isAdmin()) {
+            return ProjectRole.OWNER;
+        }
+
+        final ProjectMetadata metadata = project.metadata();
```

Comment:
Question) I believe `Project#metadata` is updated asynchronously. This means if a user changes his/her metadata and directly navigates to `/projects`, I think it is possible that the change may not be reflected depending on whether the `find` in the listener has been invoked or not.

I think it should be fine since listing "my projects" can be considered non-critical, but just wanted to make sure if this is an intentional design choice as opposed to just using a write-through cache. (I haven't checked if there are any UI paths which update metadata and navigate to `/projects` immediately)

Reply:
Good point. As you know, the consistent model of Central Dogma is eventual consistency and does not provide a transaction mechanism. 

Even before this change, users may see old content right after a commit is pushed successfully. For example, `ReplicationLagTolerantCentralDogma` is a workaround to make up for the downside of eventual consistency.

Reply:
I see, I understood that even if a user creates a project and directly navigates to the main page he/she may not see her project highlighted.

Reply:
When I tested, I was able to see newly created projects in most cases. But it may happen if there is a significant delay. Let's leave it a known issue for now.